src/main.rs

```diff
@@ -51,7 +52,7 @@ const MAX_CHUNK_LEN: u64 = 4096 * 64;
 /// The plan is tiny compared to the data and we assume we’re not dealing with
 /// malicious senders or receivers, so it doesn’t matter so much.
 #[derive(BorshSerialize, BorshDeserialize, Debug)]
-struct TransferPlan(Vec<FilePlan>);
+struct TransferPlan(u16, Vec<FilePlan>);
```

Comment:
```suggestion
struct TransferPlan {
    write_proto_version: u16,
    files: Vec<FilePlan>,
}
```
I would make the fields named to clarify what the `u16` means.